src/services/IConfiguration.hh

```diff
@@ -22,6 +22,10 @@ class IConfiguration
 {
 public:
     virtual ~IConfiguration() noexcept = default;
+	IConfiguration(const IConfiguration&) noexcept = delete;
+	IConfiguration& operator=(const IConfiguration&) noexcept = delete;
+	IConfiguration(IConfiguration&&) noexcept = delete;
+	IConfiguration& operator=(IConfiguration&&) noexcept = delete;
```

Comment:
I was copying and pasting a lot. I guess where I copied from there were tabs. I will go one by one where this happens to prevent the line count from going too high.